File: typescript-datamodelpoisoning/secure8.ts
import * as brain from "brain.js";
import express from "express";

const app = express();
app.use(express.json());

const net = new brain.NeuralNetwork();
net.train([
  { input: [0, 0], output: [0] },
  { input: [1, 1], output: [0] },
  { input: [0, 1], output: [1] },
  { input: [1, 0], output: [1] },
]);

app.post("/predict", (req, res) => {
  const result = net.run(req.body.input);
  res.json({ result });
});

app.listen(3000);

File: python-supplychain/secure8.py
import numpy as np
import onnxruntime as ort
from flask import Flask, jsonify, request

app = Flask(__name__)
session = ort.InferenceSession("./models/classifier.onnx")

@app.route("/predict", methods=["POST"])
def predict():
    data = request.json.get("features")
    input_array = np.array(data, dtype=np.float32).reshape(1, -1)
    result = session.run(None, {"input": input_array})
    return jsonify({"prediction": result[0].tolist()})
